server/commands/accountProvisioner.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+import invariant from "invariant";
+import Sequelize from "sequelize";
+import {
+  AuthenticationError,
+  EmailAuthenticationRequiredError,
+} from "../errors";
+import { User } from "../models";
+import teamCreator from "./teamCreator";
+import userCreator from "./userCreator";
+
+export default async function accountProvisioner({
+  ip,
+  user: userParams,
+  team: teamParams,
+  authenticationProvider,
+  authentication,
+}: {|
+  ip: string,
+  user: {|
+    name: string,
+    email: string,
+    avatarUrl?: string,
+  |},
+  team: {|
+    name: string,
+    domain?: string,
+    subdomain: string,
+    avatarUrl?: string,
+  |},
+  authenticationProvider: {|
+    name: string,
+    providerId: string,
+  |},
+  authentication: {|
+    providerId: string,
+    scopes: string[],
+    accessToken?: string,
+    refreshToken?: string,
+  |},
+|}) {
+  let team;
+  let isFirstUser = false;
+  try {
+    [team, isFirstUser] = await teamCreator({
+      name: teamParams.name,
+      domain: teamParams.domain,
+      subdomain: teamParams.subdomain,
+      avatarUrl: teamParams.avatarUrl,
+      authenticationProvider,
+    });
+  } catch (err) {
+    throw new AuthenticationError(err.message);
+  }
+  invariant(team, "Team must exist");
+
+  const authP = team.authenticationProviders[0];
```

Comment:
feels unsafe — this is an explicit guarantee but it violates demeter — better to use `_.find(match the provider id)`